src/bun.js/bindings/OsBinding.cpp

```diff
@@ -0,0 +1,25 @@
+#include "root.h"
+
+#if OS(DARWIN)
+#include <mach/vm_types.h>
+#include <mach/mach_host.h>
+#include <mach/mach_init.h>
+#include <mach/message.h>
+#include <mach/vm_statistics.h>
+#include <unistd.h>
+
+// Adapted from libuv darwin uv_get_free_memory, MIT
+extern "C" uint64_t Bun__Os__getFreeMemory(void)
+{
+    vm_statistics_data_t info;
```

Comment:
We don't need to worry about zero-initializing this?

Reply:
no, it's initialized by `host_statistics()`